Sources/StreamChat/Repositories/ConnectionRepository.swift

```diff
@@ -42,6 +42,10 @@ class ConnectionRepository {
         self.timerType = timerType
     }
 
+    func initialize() {
+        webSocketClient?.initialize()
+    }
```

Comment:
This needs to be introduced. Otherwise, we won't report the status of the new connection to the delegates. This is especially important when using ChatClient as a singleton. This is the problem:

1. `connectUser()` (`connectionStatus = .initialized`)
2. `timeout()` (connectinStatus = .disconnected) - Disconnected status is reported ✅ 
3. `connectUser()` again (`connectionStatus == .disconnected`) - Connecting status is not reported ❌ , because the status did not change

Since we don't reset the `connection status`, it keeps at the `.disconnected`, and so the `.connecting` or `.initialized` is not reported when manually reconnecting.

Reply:
when you call `connectUser` again, shouldn't it move to `initialized` / `connecting` already? Because there should be a status change when you start connecting again.

Reply:
I've added more details to point `1.`. The problem here is that we only set `connectionState = .connecting` after the token phase finishes. So, if the token provider keeps failing, we never set the websocket to `connecting` and it will be kept `disconnected` on the second try. And so, the connection status is not reported ever

Reply:
```mermaid
stateDiagram-v2
    [*] --> initialize
    note right of initialize
        State reported: .initialize
    end note
    
    initialize --> Connect
    Connect --> TokenProvider
    TokenProvider --> ReconnectionTimeout

     note right of TokenProvider
        State not changed to .connecting
        Because the websocket engine is not called
    end note
    
    ReconnectionTimeout --> disconnected
    note right of disconnected
        State reported: .disconnected
    end note
    
    disconnected --> Connect2
    Connect2 --> TokenProvider2
    TokenProvider2 --> ReconnectionTimeout2
    ReconnectionTimeout2 --> disconnected2
    
    note right of disconnected2
        State NOT reported
        Because it did not change from .disconnected
    end note
```

Reply:
Ideally it would be nice to see `.connecting` as the state just  after calling `connect()`. As a SDK user, who does not know internals, that would make sense. That said, it might make sense to do this separately (because we have this distinction at the moment where `.initialized` is used during the token fetch phase).

Especially because documentation says (one typo there):
```
    /// The initial state meaning that  there was no atempt to connect yet.
    case initialized
```
That is confusing for the SDK user. 

My view is that I feel like it is OK in the PR to go for `.initialized`, because this is what how the status has been reported previously, but then in a follow-up PR actually change this to `.connecting` because that would make more sense for SDK users (thinking about splitting because no idea what side-effects it could have and that should be tested throughly).

Reply:
Yes, exactly, I had the same thinking as Toomas. Basically, this part:
"The problem here is that we only set connectionState = .connecting after the token phase finishes.", doesn't feel correct. We are connecting (we are trying to get a token), but that's not reflected anywhere. To reduce risks, I'm fine to merge it like this as well, but yeah, we should make this more explicit.

Reply:
Yes, changing this would be very risky at the moment, that is why I decided not to do, and I remember I had a couple of issues trying it.

Reply:
@laevandus I changed that comment since it is internal either way. I tried changing the state to `.connecting` but this breaks other stuff, and so for now I prefer to not risk it. At most, adding another internal state, like `.fetchingToken `or something like this, would be better and not cause any impact on the rest of the logic. But for now, I think this is more than enough 👍

Reply:
Agree that we should try to tackle in another issue & PR.